Add missing locations to AzureLocation struct

jerobado opened the pull request:
Fixes #47698

jerobado committed:
Add missing locations to AzureLocation struct

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Core](https://apiview.dev/Assemblies/Review/ba87b735158144eea6cabe21a2c58dde?revisionId=d07fe02497ab4eb3b6dd64b76762fd94)

jsquire reviewed (commented):
Thanks very much for your contribution, @jerobado!   Would you be so kind as to add an item into the change log that captures the additions?

jerobado commented:
Thanks @jsquire for the review.

I'll update the PR within the day.

jerobado committed:
Update CHANGELOG.md

jsquire reviewed (approved):
Thanks, @jerobado!   I'll get this merged  in.

jsquire commented:
@jerobado:  Seems that I overlooked the build failure.   We'll need to run a PowerShell script to update the API listings, which should unblock.

`eng\scripts\Export-API.ps1 -ServiceDirectory core`

jerobado committed:
Regenerate API listing via PowerShell scripts

These files are generated by running the following scripts:

`.\eng\scripts\Export-API.ps1 -ServiceDirectory core  && .\eng\scripts\Update-Snippets.ps1 - ServiceDirectory core`
